index.js

```diff
@@ -59,6 +86,7 @@ module.exports = function(fn) {
 
     res.end = function(chunk, encoding, cb) {
       debug('end called');
+      endCalled = true;
```

Comment:
Maybe you could also add an example addressing this case.

> The only drawback to this approach is that I cannot figure out how to support the custom status messages, but that seemed like an OK compromise to me.

What do you think about this guys @flockonus @papandreou @David-Hari ?